### PR TITLE
fix: check webpack installation before running cli

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
fix

**Did you add tests for your changes?**
no
**If relevant, did you update the documentation?**
no
**Summary**
prompt to install `webpack` if it's not present already.

Screenshot at 2020-09-24 13-55-05

**Does this PR introduce a breaking change?**
No

**Other information**
Refers https://github.com/webpack/webpack-cli/issues/1383#issuecomment-608225508
Fixes https://github.com/webpack/webpack-cli/issues/1383

### CODE REVIEW

File: packages/webpack-cli/lib/utils/cli-flags.js [hunk 1]
```diff
@@ -8,6 +8,11 @@ const ADVANCED_GROUP = 'advanced';
 const DISPLAY_GROUP = 'stats';
 const ZERO_CONFIG_GROUP = 'zero-config';
 
+let cli = undefined;
+if (packageExists('webpack')) {
+    cli = require('webpack').cli;
+}
```
Comment (evilebottnawi) [inline packages/webpack-cli/lib/utils/cli-flags.js:14]: Why here?

File: packages/webpack-cli/bin/cli.js [hunk 2]
```diff
@@ -4,13 +4,21 @@
 require('v8-compile-cache');
 const importLocal = require('import-local');
 const runCLI = require('../lib/bootstrap');
+const { yellow } = require('colorette');
+const { packageExists, promptInstallation } = require('@webpack-cli/package-utils');
```
Comment (evilebottnawi) [inline packages/webpack-cli/bin/cli.js:8]: Do we add `package-utils` in deps?

**Review by evilebottnawi — COMMENTED** [review]
Good idea!

File: packages/webpack-cli/bin/cli.js [hunk 3]
```diff
@@ -4,13 +4,21 @@
...
 // Prefer the local installation of webpack-cli
 if (importLocal(__filename)) {
     return;
 }
 process.title = 'webpack';
 
-const [, , ...rawArgs] = process.argv;
-
-runCLI(rawArgs);
+if (packageExists('webpack')) {
```
Comment (ylemkimon) [inline packages/webpack-cli/bin/cli.js:16]: For `@webpack-cli/package-utils` to detect whether `webpack` exists, `webpack` should be listed as a peer dependency of `@webpack-cli/package-utils`, or at least as an optional peer dependency via `peerDependenciesMeta` (#1816).

File: packages/webpack-cli/bin/cli.js (hunk 2 above)
Comment (snitin315) [reply packages/webpack-cli/bin/cli.js:8]: Yes Already present.

https://github.com/webpack/webpack-cli/blob/5bae7977949f0e78df60e795eae171a17f4677e4/packages/webpack-cli/package.json#L26

File: packages/webpack-cli/lib/utils/cli-flags.js (hunk 1 above)
Comment (snitin315) [reply packages/webpack-cli/lib/utils/cli-flags.js:14]: Can be refactored as -> `const cli = packageExists('webpack') ? require('webpack').cli : undefined`

Else we get `Error: can't find module 'webpack'`

File: packages/webpack-cli/bin/cli.js (hunk 2 above)
Comment (snitin315) [reply packages/webpack-cli/bin/cli.js:8]: But these utils are only being used by `webpack-cli`, So we can safely move it from `package-utils` to `utils`. Let's do this in separate PR.

File: packages/webpack-cli/bin/cli.js [hunk 4]
```diff
@@ -4,13 +4,21 @@
...
+    const [, , ...rawArgs] = process.argv;
+    runCLI(rawArgs);
+} else {
+    promptInstallation('webpack', () => {
+        console.error(`It looks like ${yellow('webpack')} is not installed.`);
```
Comment (jamesgeorge007) [inline packages/webpack-cli/bin/cli.js:21]: Use `logger`?

File: packages/webpack-cli/bin/cli.js (hunk 2 above)
Comment (anshumanv) [reply packages/webpack-cli/bin/cli.js:9]: https://github.com/webpack/webpack-cli/pull/1822 will solve this

File: packages/webpack-cli/bin/cli.js (hunk 4 above)
Comment (snitin315) [reply packages/webpack-cli/bin/cli.js]: done

**Review by evilebottnawi — APPROVED** [review]

**Review by rishabh3112 — APPROVED** [review]

**Review by anshumanv — APPROVED** [review]

**Pull request merged**